scripts/lib/dev-secrets.sh

```
@@ -129,7 +180,18 @@ cmux_dev_secrets_load() {
     case "$1" in EMAIL) cmux_dev_secrets__read_key "$agent_file" CMUX_UITEST_STACK_EMAIL ;; PASSWORD) cmux_dev_secrets__read_key "$agent_file" CMUX_UITEST_STACK_PASSWORD ;; esac
   }
 
-  if [[ "$agent_only" -eq 1 ]]; then
+  if [[ -n "$explicit_file" ]]; then
+    # An explicit one-shot file is exclusive. Never fall through to ambient
+    # development credentials during a production release gate.
+    cmux_dev_secrets__explicit_dogfood() {
+      case "$1" in EMAIL) cmux_dev_secrets__read_key "$explicit_file" CMUX_DOGFOOD_STACK_EMAIL ;; PASSWORD) cmux_dev_secrets__read_key "$explicit_file" CMUX_DOGFOOD_STACK_PASSWORD ;; esac
+    }
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)